src/scheduler/EventOrganizer.java

```diff
@@ -65,43 +66,62 @@ public void addEvent(String commandAdd) {
             // Check if the date is in a valid format
             if (!eventDate.isValid()) {
                 System.out.println(
-                        "Invalid date format. Please use mm/dd/yyyy format.");
+                        eventDate + " Invalid date format.");
                 return;
             }
 
             // Check if the date is more than 6 months away from the current
             // date
             if (eventDate.isMoreThanSixMonthsAway()) {
                 System.out.println(
-                        "Event date is more than 6 months away from today’s " +
-                        "date.");
+                        eventDate + ": Event date must be within 6 months!");
                 return;
             }
 
             // Check if the date is a future date
             if (!eventDate.isFutureDate()) {
-                System.out.println("Event date should be a future date.");
+                System.out.println( eventDate + ":Event date should be a future date.");
                 return;
             }
 
             Timeslot timeSlot =
                     Timeslot.valueOf(tokens[TIMESLOT_TOKEN_INDEX].toUpperCase());
-            Location location =
-                    Location.valueOf(tokens[LOCATION_TOKEN_INDEX].toUpperCase());
+            String locationToken = tokens[LOCATION_TOKEN_INDEX].toUpperCase();
+
+            if (Location.isValidLocation(locationToken)) {
```

Comment:
isValidLocation returns true if not valid?

Reply:
Correct

Reply:
this is the source of your location bug. Also get rid of the semi colon after the if statement's closing bracket